ckanext/dcatapchharvest/dcat_helpers.py

```diff
@@ -153,3 +154,23 @@ def get_theme_mapping():
                                  predicate=SKOS.mappingRelation)
             if g.namespace_manager.compute_qname(obj)[0] == 'euthemes']
     return theme_mapping
+
+
+def get_pagination(catalog_graph):
+    pagination = {}
+    for pagination_node in catalog_graph.subjects(
+            RDF.type, HYDRA.PagedCollection
+    ):
+        items = [
+            ('next', HYDRA.nextPage),
+            ('first', HYDRA.firstPage),
+            ('last', HYDRA.lastPage),
+            ('count', HYDRA.totalItems),
+            ('items_per_page', HYDRA.itemsPerPage),
+        ]
+        for key, ref in items:
+            for obj in catalog_graph.objects(pagination_node, ref):
+                pagination[key] = unicode(obj)
+                log.error("{} {}".format(key, unicode(obj)))
+    log.error(pagination)
```

Comment:
question: is this logging here intentional? if so, is it really representing an error here? maybe remove this line or change it to `info` with a message that is more useful?

Reply:
I choose this because the harvester errors where also logged this way in other parts of the code, see here for example: https://github.com/opendata-swiss/ckanext-dcatapchharvest/blob/7a9df1d66bbc72c0fd98b6e9e07700e4ca0dd3ff/ckanext/dcatapchharvest/harvesters.py#L94 

I can change it to log.info() but what about these other harvester errors, should they also be reported as info rather then as error? If you have an opinion on that I could change those as well. I think it might be a matter of taste on how to log the harvester errors. But I think you are right and will at least these ones change to Info. Good point.

Reply:
Hi! Sorry to comment on this a week after it's been merged, but I just saw this and wanted to mention that I don't think this way of logging errors is helpful at all.

It looks like we call `get_pagination()` every time `after_parsing()` is called, so after parsing every page of the source data. It can't be the case that `pagination` is an error every time, so we definitely should not be logging it as an error. We also log this as `debug` in `after_parsing()`. To my reading, it looks like the information we would log for every single page would be something like this:

```
ERROR: next (some number)
ERROR: first (some number)
ERROR: last (some number)
ERROR: count (some number)
ERROR: items_per_page (some number)
ERROR: {next: (some number), first: (some number), last: (some number), count: (some number), items_per_page: (some number)}
DEBUG: pagination-info: {next: (some number), first: (some number), last: (some number), count: (some number), items_per_page: (some number)}
```

This is messy. Most of these lines also do not include any information about what this data is or why we would log it at all.

If there is an error found that prevents gathering the data, what we should do imo is to save a gather error using the method for that in the harvester, with an error message that explains what happened. This ensures that the error is reported to the user, so they know what happened and can take steps to fix their data. Using `log.error()` only logs the messages to the CKAN log in the cloud, which means they will only be visible to hosting support or to us devs if we turn on Sentry logging.

This is actually what is done just below the line you referenced: https://github.com/opendata-swiss/ckanext-dcatapchharvest/blob/7a9df1d66bbc72c0fd98b6e9e07700e4ca0dd3ff/ckanext/dcatapchharvest/harvesters.py#L95

Looking at the git history of the code, I believe that the lines in `_get_guid()` where we save an error message with `log.error()` were added either because, at that time, we were running CKAN on a server where we devs had easy access to the logs, or for convenience while developing and not deleted before the code was merged in.

I like the approach you took here, using the new interface of the DCAT harvester, and I would really appreciate it if you could fix the code here so that gather errors are saved properly, instead of logging a lot of data that we can't see or use. Thank you!